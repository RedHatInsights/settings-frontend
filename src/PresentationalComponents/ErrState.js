import React from 'react';
import propTypes from 'prop-types';
import ErrorState from '@redhat-cloud-services/frontend-components/ErrorState';

export const ErrState = ({ errorTitle, errorDescription }) => {
  return (
    <ErrorState
      errorTitle={errorTitle}
      errorDescription={
        <div dangerouslySetInnerHTML={{ __html: errorDescription }} />
      }
    />
  );
};

ErrState.propTypes = {
  errorTitle: propTypes.string,
  errorDescription: propTypes.string,
};
