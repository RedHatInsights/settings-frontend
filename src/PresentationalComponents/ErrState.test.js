import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ErrState } from './ErrState';

it('plain text hack - render', () => {
  const wrapper = shallow(
    <ErrState errorTitle="Testing" errorDescription="This is a test" />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
