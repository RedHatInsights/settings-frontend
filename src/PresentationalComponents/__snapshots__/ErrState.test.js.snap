// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plain text hack - render 1`] = `
<ErrorState
  errorDescription={
    <span>
       
      <br />
       
      <a />
    </span>
  }
  errorTitle="Something went wrong"
/>
`;
